### PR TITLE
feat(hybridcloud) Trigger separate silo DB tests with labels

### PR DESCRIPTION
**Pull request by markstory**
We're at the point now where foreign keys have been broken and the migration history with foreign keys has been squashed. We can now generate new environments using siloed database connections to futher validate that application logic isn't crossing silos without RPC calls.

I thought that using a label to trigger using split databases would be an unobtrusive way for the hybrid cloud team to work incrementally on increasing silo stability. Once we have all tests passing running tests against split databases will likely become the new normal, as any logic that works under split databases will also work under a shared database.

I've not created the proposed label yet, as I wanted to get feedback on this approach before adding a label.

Refs HC-769

### CODE REVIEW

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -384,6 +409,12 @@ jobs:
           # https://github.com/codecov/codecov-bash/blob/7100762afbc822b91806a6574658129fe0d23a7d/codecov#L891
           fetch-depth: '2'
 
+      - name: Update environment for silo databases
+        if: |
+          contains( github.event.pull_request.labels.*.name, 'Trigger: Silo db' )
+        run: |
+          echo '::set-env name=SENTRY_USE_SILOS=1'
```
Comment (asottile-sentry) [inline .github/workflows/backend.yml:416]: these `set-env` commands haven't worked for months -- you're looking for `$GITHUB_ENV` https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

File: .github/workflows/backend.yml [hunk 2]
```diff
@@ -88,6 +88,12 @@ jobs:
           # https://github.com/codecov/codecov-bash/blob/7100762afbc822b91806a6574658129fe0d23a7d/codecov#L891
           fetch-depth: '2'
 
+      - name: Update environment for silo databases
+        if: |
+          contains( github.event.pull_request.labels.*.name, 'Trigger: Silo db' )
+        run: |
+          echo "SENTRY_USE_SILOS=1" >> "$GITHUB_ENV"
```
Comment (asottile-sentry) [inline .github/workflows/backend.yml:95]: should we put this in `setup-sentry` directly?

Comment (markstory) [reply .github/workflows/backend.yml]: I don't know. I'm hoping this is a short-lived label based workflow. Once we have master green with split databases, I'd like to make it the default behavior as it won't add overhead and will prevent new cross db operations from being introduced.

**Comment by markstory** [comment]
For more context on how this would be used. I've done some of the ground work to get CI running against split databases in #51307. Once those changes are merged, the plan would be to make incremental fixes for the ~675 tests that are failing, using the labels to validate our progress.

**Comment by markstory** [comment]
@asottile-sentry Anything else I need to do on this?

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**